### PR TITLE
fix bad simulation ownership when using edit.js

### PR DESCRIPTION
**Pull request by AndrewMeadows**
This PR fixes some bad ownership when using edit.js.  It might also fix some of the simulation flickering seen when multiple interfaces simulate tumbling objects.

### CODE REVIEW

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -628,72 +628,51 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
...
+    READ_ENTITY_PROPERTY(PROP_GRAVITY, glm::vec3, updateGravity);
+
+    READ_ENTITY_PROPERTY(PROP_DAMPING, float, updateDamping);
+    READ_ENTITY_PROPERTY(PROP_RESTITUTION, float, updateRestitution);
+    READ_ENTITY_PROPERTY(PROP_FRICTION, float, updateFriction);
+    READ_ENTITY_PROPERTY(PROP_LIFETIME, float, updateLifetime);
+    READ_ENTITY_PROPERTY(PROP_SCRIPT, QString, setScript);
+    READ_ENTITY_PROPERTY(PROP_SCRIPT_TIMESTAMP, quint64, setScriptTimestamp);
+    READ_ENTITY_PROPERTY(PROP_REGISTRATION_POINT, glm::vec3, setRegistrationPoint);
+
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: This diff looks complicated, but what is really happening is I'm finally getting around to removing a bunch of cruft for backward version compatibility when the simulation ownership stuff was first added.  These are not the interesting changes you are looking for.

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -1104,7 +1072,7 @@ bool EntityItem::setProperties(const EntityItemProperties& properties) {
     bool somethingChanged = false;
 
     // these affect TerseUpdate properties
-    SET_ENTITY_PROPERTY_FROM_PROPERTIES(simulationOwner, setSimulationOwner);
+    SET_ENTITY_PROPERTY_FROM_PROPERTIES(simulationOwner, updateSimulationOwner);
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: This is bug #2: simulation ownership on the server was not always setting the DIRTY_SIMULATOR_ID flag, which is used by the **SimpleEntitySimulation** to remember to put it on a list of owned objects.  The simulation owner would be properly set, but the server would not remember to clear out the ownership when the owner logged off.

File: libraries/entities/src/EntityItem.cpp [hunk 3]
```diff
@@ -1324,6 +1292,13 @@ void EntityItem::updatePosition(const glm::vec3& value) {
...
     if (getLocalPosition() != value) {
         setLocalPosition(value);
+        _dirtyFlags |= Simulation::DIRTY_POSITION;
+        forEachDescendant([&](SpatiallyNestablePointer object) {
+            if (object->getNestableType() == NestableTypes::Entity) {
+                EntityItemPointer entity = std::static_pointer_cast<EntityItem>(object);
+                entity->_dirtyFlags |= Simulation::DIRTY_POSITION;
+            }
+        });
     }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: This is the fix for bug number 1: the **DIRTY_POSITION** flag was not being set on objects who had suffered position updates from outside the simulation --> the **PhysicsEngine** would not update the position of the corresponding **RigidBody**.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 987b8d3334a1136dbce95122159a98650f4216b7 available at: https://gist.github.com/6d39237ff744d7569348<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6596.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6596.dmg

**Pull request merged**